### PR TITLE
Eos 20364 health tabular view

### PR DESCRIPTION
**Pull request by Vrishali-web-dev**
# Frontend

# Problem Statement

- _Overview: JIRA number/GitHub Issue added to PR: [Y/N]:_
- _Describe the problem this patch intends to solve._

## Design

- _For Bug describe the fix here._
- _For feature, post the link to the solution page._
- _Any northbound interface change and has been notified to other gatekeepers [Y/N]:_

## Coding

- _Coding conventions are followed and code is consistent. Yes/No?_
- _Confirm All CODACY errors are resolved. Yes/No?_

## Testing

- _Confirm that Test Cases are added (for both the cases, fix and feature). Yes/No?_
- _Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability. Yes/No?_
- _Confirm Testing was performed with installed RPM. Yes/No?_
- _Confirm all the ut/regression/smoke test has been performed [Y/N]_

## Checklist

- _PR is self-reviewed? Yes/No?_
- _GitHub Issue is updated_
- _Jira is updated_
  -  _Check if the description is clear and explained._ 
    -  _Check Acceptance Criterion is defined._
      -  _All the tests performed should be mentioned before Resolving a JIRA._
        -  _Verification needs to be done before marked as Closed/Verified_
        - _Any interface change Yes/No?_
        - _Change notified to other gatekeepers: Yes/No?_
        - _Code reviews done and addressed: Yes/No?_
        - _Codacy issues reviewed and addressed: Yes/No?_
        - _Deployment test : Done/Not?_
        - _UT/smoke test: Done/Not?_
        - _Jira number added to PR: Yes/No?_
        - _Github merge done using UI: Yes/No?_
        - _Side effects on other features - deployment/update/node-replacement_
        - _Changes to quick-start-guide/community impact_
        - _All dependent component code PR are raised or already merged Yes/No_
        - _All dependent code already merged in landing branch Yes/No_

-----
[View rendered .github/pull_request_template.md](https://github.com/Vrishali-web-dev/cortx-management-portal/blob/EOS-20364-health-tabular-view/.github/pull_request_template.md)

### CODE REVIEW

File: gui/src/components/health/cortx-health-tabular.vue [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  }  
+
+  public async getHealthData() {
+    this.healthQueryParams = {
+      response_format: 'flattened',
+      offset: 1,
+      limit: 0
+    }
+    this.$store.dispatch("systemConfig/showLoader", "Table in progress...");
+    try {
```
Comment (shrimetta) [inline gui/src/components/health/cortx-health-tabular.vue:62]: try catch is not required. We have global error handling in place.

File: gui/src/components/health/cortx-health-tabular.vue [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    this.getHealthData();    
+  }  
+
+  public async getHealthData() {
+    this.healthQueryParams = {
+      response_format: 'flattened',
+      offset: 1,
+      limit: 0
+    }
+    this.$store.dispatch("systemConfig/showLoader", "Table in progress...");
```
Comment (shrimetta) [inline gui/src/components/health/cortx-health-tabular.vue:61]: Message should be "Fetching health..."

File: gui/src/components/health/cortx-health.vue [hunk 3]
```diff
@@ -15,15 +15,61 @@
 * please email opensource@seagate.com or cortx-questions@seagate.com.
 */
 <template>
-  <router-view class="cortx-p-1"></router-view>
+  <div class="cortx-p-1">
+    <cortx-tabs :tabsInfo="tabsInfo" />
+      <CortxHealthGraphical v-if="showGraphTab" />
```
Comment (shrimetta) [inline gui/src/components/health/cortx-health.vue:20]: Please remove graphical tab as it is not implemented yet.

File: web/src/services/api-register.ts [hunk 4]
```diff
@@ -64,6 +64,7 @@ export default {
   get_product_version_endpt: (version :string) => `/api/${version}/product_version`,
   get_health_components_endpt: (version :string) => `/api/${version}/system/health/components`,
   get_health_resources_endpt: (version :string) => `/api/${version}/system/health/resources`,
+  get_health_cluster_endpt: (version :string) => `/api/${version}/system/health/node`,
```
Comment (shrimetta) [inline web/src/services/api-register.ts:67]: No need to hard code node here. You can use another parameter for resource.
Ex: get_health_cluster_endpt: (version :string, resource: string) => `/api/${version}/system/health/${resource}`

**Review by shrimetta — CHANGES_REQUESTED** [review]
Some minor comments. Please address.

File: gui/src/components/health/cortx-health-tabular.vue (hunk 1 above)
Comment (Vrishali-web-dev) [reply gui/src/components/health/cortx-health-tabular.vue]: Updated

File: gui/src/components/health/cortx-health-tabular.vue (hunk 2 above)
Comment (Vrishali-web-dev) [reply gui/src/components/health/cortx-health-tabular.vue]: Text is updated

File: gui/src/components/health/cortx-health.vue (hunk 3 above)
Comment (Vrishali-web-dev) [reply gui/src/components/health/cortx-health.vue]: Removed Graph tab

File: web/src/services/api-register.ts (hunk 4 above)
Comment (Vrishali-web-dev) [reply web/src/services/api-register.ts]: Updated endpoint implementation

File: gui/src/services/api-register.ts [hunk 5]
```diff
@@ -69,6 +69,7 @@ export default {
   health_components: `api/${version}/system/health/components`,
   system_status: `/api/${version}/system/status`,
   health_resources: `api/${version}/system/health/resources`,
+  health_cluster: `api/${version}/system/health/node`,
```
Comment (shrimetta) [inline gui/src/services/api-register.ts:72]: No need of hard coding node here. It is only required at GUI. You can pass the value that you are getting from gui here.

Comment (shrimetta) [reply gui/src/services/api-register.ts:72]: Added the comment by mistake

**Review by shrimetta — APPROVED** [review]

**Pull request merged**